Core/clim-core/frames.lisp

```diff
@@ -1366,7 +1366,41 @@ have a `pointer-documentation-pane' as pointer documentation,
 
 (defun frame-display-pointer-documentation-string (frame string)
   (with-output-to-pointer-documentation (stream frame)
-    (write-string string stream)))
+      (write-string string stream))
+  (let ((*pointer-documentation-output* (frame-pointer-documentation-output frame)))
+    ;; To see the string it is necessary to trigger the redisplay of
+    ;; pointer-documentation-pane with FRAME-UPDATE-POINTER-DOCUMENTATION.
+    ;; I don't know why we need to trigger the update function twice to see
```

Comment:
I think we need to first understand why it is needed to be called twice (probably that will reveal an issue somewhere else).

Reply:
> I think we need to first understand why it is needed to be called twice (probably that will reveal an issue somewhere else).

I think I found a missing `force-output` around the pointer documentation code when working on the fb backend. Maybe that's the problem?

Reply:
why not `finish-output`?

Reply:
Actually the two calls of `FRAME-UPDATE-POINTER-DOCUMENTATION` are necessary only if you call `FRAME-DISPLAY-POINTER-DOCUMENTATION-STRING` from SLIME. If you call the function from a CLIM application one call is enough also if you send a string on a different application-frame. So we can:
1. continue to investigate why from `slime` two calls are necessary.
2. remove one of the two calls (`FRAME-DISPLAY-POINTER-DOCUMENTATION-STRING` is supposed to be called from CLIM not from SLIME).
3. leave the two calls

Reply:
currently stream access from outside clim loop is prone to race conditions and it is a known problem, so I think it should be just a single call. there is a ticket for stream concurrent access so there is no need for a separate issue either.